src/prng/chacha.rs

```diff
@@ -211,39 +237,11 @@ impl Rng for ChaChaRng {
         ::rand_core::impls::next_u128_via_u64(self)
     }
     
-    // Custom implementation allowing larger reads from buffer is about 8%
-    // faster than default implementation in my tests
     fn fill_bytes(&mut self, dest: &mut [u8]) {
-        use core::cmp::min;
-        use core::intrinsics::{transmute, copy_nonoverlapping};
-        
-        let mut left = dest;
-        while left.len() >= 4 {
-            if self.index == STATE_WORDS {
-                self.update();
-            }
-            
-            let words = min(left.len() / 4, STATE_WORDS - self.index);
-            let (l, r) = {left}.split_at_mut(4 * words);
-            left = r;
-            
-            // convert to LE:
-            for ref mut x in self.buffer[self.index..self.index+words].iter_mut() {
-                **x = w((*x).0.to_le());
```

Comment:
I don't think mutating the output buffer from ChaCha is correct... The output buffer is used for the next round. Have to think of something else.

I have tried to set up big-endian testing several times, no success yet. Do you happen to know an easy way (that works on Fedora)?

Reply:
Oops, no it doesn't

Reply:
I guess you need an emulator. If you like I can try to set something up.

https://stackoverflow.com/questions/3337896/imitate-emulate-a-big-endian-behavior-in-c

Reply:
Do you think we could get CI? With [trust](https://github.com/japaric/trust) we would get big-endian testing for free. I have not made a pr yet, but last I checked we didn't build on Windows (small fix though).